code/__HELPERS/reagents.dm

```diff
@@ -72,3 +72,14 @@
 	if(!GLOB.chemical_reactions_list[primary_reagent])
 		GLOB.chemical_reactions_list[primary_reagent] = list()
 	GLOB.chemical_reactions_list[primary_reagent] += R
+
+/proc/create_foam(datum/effect_system/foam_spread/foam,datum/reagents/holder,foam_volume,metaltype = 0,notification="<span class='danger'>The solution spills out foam!</span>")
+	var/location = get_turf(holder.my_atom)
+	for(var/mob/M in viewers(5, location))
+		to_chat(M, notification)
+	if(metaltype == 0)
```

Comment:
Would be better to clean up the parameters of base foam set_up instead of needing this if.